src/EnergyPlus/WaterThermalTanks.hh

```diff
@@ -69,86 +69,185 @@ struct EnergyPlusData;
 
 namespace WaterThermalTanks {
 
-    enum struct AmbientTempEnum
+    enum class WTTAmbientTemp
     {
+        Unassigned = -1,
         Schedule,   // ambient temperature around tank (or HPWH inlet air) is scheduled
         TempZone,   // tank is located in a zone or HPWH inlet air is zone air only
         OutsideAir, // tank is located outdoors or HPWH inlet air is outdoor air only
-        ZoneAndOA   // applicable to HPWH only, inlet air is mixture of OA and zone air
+        ZoneAndOA,  // applicable to HPWH only, inlet air is mixture of OA and zone air
+        Num
     };
 
-    enum struct CrankTempEnum
+    constexpr std::array<std::string_view, static_cast<int>(WTTAmbientTemp::Num)> HPWHAmbientTempNamesUC{
+        "SCHEDULE", "ZONEAIRONLY", "OUTDOORAIRONLY", "ZONEANDOUTDOORAIR"};
+
+    constexpr int TankAmbientTempNamesNum = static_cast<int>(WTTAmbientTemp::Num) - 1; // Since AmbientTemp::ZoneAndOA is not appilcable to Tank
```

Comment:
I would define this as another enumerated value in WTTAmbientTemp:
```cpp
    Num,
    TankNum = Num - 1
};
```

Reply:
Will change, thanks!